src/platform/Darwin/BleConnectionDelegateImpl.mm

```diff
@@ -375,34 +375,30 @@ - (void)peripheral:(CBPeripheral *)peripheral
     didUpdateValueForCharacteristic:(CBCharacteristic *)characteristic
                               error:(NSError *)error
 {
+    auto bridgedPeripheral = (__bridge void *) peripheral;
```

Comment:
How does the refcounting here work?  When the async block was using `peripheral`, it ensured that it stayed alive.  But now that it only loses `bridgedPeripheral`, which is not a strong ref, what makes sure that `peripheral` does not die on us?

Reply:
I was sort of expecting it to be kept around by the `_peripheral = peripheral;` line in `connect` (https://github.com/project-chip/connectedhomeip/blob/fc0e4a2c698b06a6467a4969981e3dbd87a82063/src/platform/Darwin/BleConnectionDelegateImpl.mm#L449)

But that is right, it could be a race. I have kept the bridge conversion as before.